Automattic/woocommerce-payments
# Add cart item count to fraud metadata

#### Changes proposed in this Pull Request

This PR adds the cart item count to fraud metadata, to let the server use the value instead of level 3 line items array, because it is only available for US merchants. This way we won't have a dependency to Level 3 data existence, and this fix will apply to all our merchants. 

Other than the main goal of this PR, this fixes several things on the frontend:

- Filters out "allow" outcomes from the block/review timeline items, so they don't cause an exception because the text mappings doesn't exist for them.
- Improves the strings used for each rule failure a bit. May need review and improvements.
- The UI was accepting amount inputs as floats, but the rule engine checks the amounts in cents, so this adds a multiplication of 100 when building the ruleset, and division of 100 when reading from a ruleset to display in the UI.









#### Testing instructions





* Tests should pass, and the changes should make sense. 
* The tests were already written for the use cases, so I modified them to match the modifications.

-------------------


**Post merge**
